tests/fast-integration/eventing-test/eventing-test-prep.js

```diff
@@ -38,47 +41,98 @@ const {
+    }
+
+    if (!skrInstanceId) {
+      info(`Skipping fetching SKR kubeconfig because skrInstanceId is not set.`);
+      this.skip();
+    }
+
+    // 'skr-test/helpers' initializes KEB clients on import, that is why it is imported only if needed
+    const {getSKRConfig} = require('../skr-test/helpers');
+    const {initK8sConfig} = require('../skr-test/helpers');
+
+    debug(`Fetching SKR config for Instance Id: ${skrInstanceId}`);
+    const shoot = await getSKRConfig(skrInstanceId);
+
+    debug('Initiating SKR K8s config...');
+    await initK8sConfig(shoot);
+  });
 
+  it('Prepare EventMesh secret', async function() {
     // If eventMeshSecretFilePath is specified then create a k8s secret for eventing-backend
-    // else use existing k8s secret as specified in backendK8sSecretName & backendK8sSecretNamespace
-    if (eventMeshSecretFilePath) {
-      debug('Creating Event Mesh secret');
-      const eventMeshInfo = await createEventingBackendK8sSecret(
-          eventMeshSecretFilePath,
-          backendK8sSecretName,
-          backendK8sSecretNamespace,
-      );
-      setEventMeshSourceNamespace(eventMeshInfo['namespace']);
+    // else skip this step and use existing k8s secret as specified in backendK8sSecretName & backendK8sSecretNamespace
+    if (!eventMeshSecretFilePath) {
+      this.skip();
+    }
+
+    debug('Creating Event Mesh secret');
+    const eventMeshInfo = await createEventingBackendK8sSecret(
+        eventMeshSecretFilePath,
+        backendK8sSecretName,
+        backendK8sSecretNamespace,
+    );
+    setEventMeshSourceNamespace(eventMeshInfo['namespace']);
+  });
+
+  it('Prepare assets without Compass flow', async function() {
+    // Skip this step if compass flow is enabled
+    if (testCompassFlow) {
+      this.skip();
     }
 
     // Deploy Commerce mock application, function and subscriptions for tests
-    if (isSKR) {
-      await prepareAssetsForSKRTests();
-    } else {
-      await prepareAssetsForOSSTests();
+    await prepareAssetsWithoutCompassFlow();
+  });
+
+  it('Prepare assets with Compass flow', async function() {
+    // Skip this step if compass flow is disabled
+    if (!testCompassFlow) {
+      this.skip();
     }
+
+    // Deploy Commerce mock application, function and subscriptions for tests (includes compass flow)
+    await prepareAssetsWithCompassFlow();
   });
 
-  // prepareAssetsForOSSTests - Sets up CommerceMost for the OSS
-  async function prepareAssetsForOSSTests() {
-    debug('Preparing CommerceMock test fixture on Kyma OSS');
+  afterEach(async function() {
+    // if the test preparation failed, perform the cleanup
+    if (this.currentTest.state === 'failed') {
+      await cleanupTestingResources();
+    }
+  });
+
+  // // **** Helper functions ****
+  // prepareAssetsWithoutCompassFlow - Sets up test assets without compass flow
+  async function prepareAssetsWithoutCompassFlow() {
+    debug('Preparing CommerceMock/In-cluster test fixtures on Kyma');
```

Comment:
Why do we need the commerce mock in the scenario without compass ?

Reply:
the preparation step still deploys the commerce mock pod and apirules, and test some apirule access. But yes, ideally we need to cleanup the code and remove commerce mock flow.

Reply:
Opened a follow-up: https://github.com/kyma-project/kyma/issues/15244